src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -354,8 +386,57 @@ public JSONObject getDebugState()
     public abstract void addRtpExtension(RtpExtension rtpExtension);
 
     /**
-     *
      * @param newVideoConstraints
      */
-    public abstract void setVideoConstraints(Map<String, VideoConstraints> newVideoConstraints);
+    public abstract void setSenderVideoConstraints(ImmutableMap<String, VideoConstraints> newVideoConstraints);
+
+    /**
+     * Notifies this instance that the max video constraints that the bridge
+     * needs to receive from this endpoint has changed. Each implementation
+     * handles this notification differently.
+     *
+     * @param maxVideoConstraints the max video constraints that the bridge
+     *                            needs to receive from this endpoint
+     */
+    protected abstract void
+    onMaxReceiverVideoConstraintsChanged(VideoConstraints maxVideoConstraints);
```

Comment:
nit: for these types of methods I think we don't use the `on` prefix, so just `maxReceiverVideoConstraintsChanged` (a method with an `on` would be used when adding a handler for an event, and the event occurring itself is without the `on`).

Reply:
cool, I renamed a few more methods.